presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -69,6 +71,10 @@ with
     : WITH RECURSIVE? namedQuery (',' namedQuery)*
     ;
 
+withOrWithoutData
```

Comment:
This is used in a single place and it doesn't result in the creation of an independent tree node, so inline it into the rule that uses it.